src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -340,18 +308,61 @@ private static double calculatePower(DcLoadFlowContext loadFlowContext, LfBranch
+                                                           PropagatedContingency contingency, Map<String, ComputedContingencyElement> contingencyElementByBranch,
+                                                           Set<LfBus> disabledBuses, List<ParticipatingElement> participatingElements, Set<String> elementsToReconnect,
+                                                           SensitivityResultWriter resultWriter, ReportNode reportNode, Set<LfBranch> partialDisabledBranches, boolean rhsChangedAfterConnectivityBreak) {
+        List<LfSensitivityFactor<DcVariableType, DcEquationType>> factors = validFactorHolder.getFactorsForContingency(contingency.getContingency().getId());
+        if (factors.isEmpty()) {
+            // no factor no more computation
+            resultWriter.writeContingencyStatus(contingency.getIndex(), SensitivityAnalysisResult.Status.FAILURE);
+            return;
+        }
+
+        Collection<ComputedContingencyElement> contingencyElements = contingency.getBranchIdsToOpen().keySet().stream()
+                .filter(element -> !elementsToReconnect.contains(element))
+                .map(contingencyElementByBranch::get)
+                .collect(Collectors.toList());
+
+        var lfNetwork = loadFlowContext.getNetwork();
+        Set<LfBranch> disabledBranches = contingency.getBranchIdsToOpen().keySet().stream().map(lfNetwork::getBranchById).collect(Collectors.toSet());
+        disabledBranches.addAll(partialDisabledBranches);
+        DisabledNetwork disabledNetwork = new DisabledNetwork(disabledBuses, disabledBranches);
+        DenseMatrix newFactorStates = factorStates;
+
         if (contingency.getGeneratorIdsToLose().isEmpty() && contingency.getLoadIdsToLoose().isEmpty()) {
-            calculateSensitivityValues(loadFlowContext, factors, factorStates, contingenciesStates, flowStates, contingencyElements,
+            DenseMatrix newFlowStates = flowStates;
+            // we need to recompute the factor states because the connectivity changed
+            if (rhsChangedAfterConnectivityBreak) {
+                newFactorStates = calculateFactorStates(loadFlowContext, factorGroups, participatingElements);
+            }
+
+            // get the lost phase tap changers for this contingency
+            Set<LfBranch> lostPhaseControllers = contingency.getBranchIdsToOpen().keySet().stream()
+                    .filter(element -> !elementsToReconnect.contains(element))
+                    .map(contingencyElementByBranch::get)
+                    .map(ComputedContingencyElement::getLfBranch)
+                    .filter(LfBranch::hasPhaseControllerCapability)
+                    .collect(Collectors.toSet());
+
+            // if a phase tap changer is lost or if the connectivity have changed, we must recompute load flows
+            if (!disabledBuses.isEmpty() || !lostPhaseControllers.isEmpty()) {
```

Comment:
Note that I have removed what you did new DisabledNetwork(disabledBuses, disabledPhaseTapChangers), I don't understand why we need it.

Reply:
Indeed, it is already covered in
DisabledNetwork disabledNetwork = new DisabledNetwork(disabledBuses, disabledBranches);

Thanks!